internal/pkg/stream/stream.go

```diff
@@ -60,3 +62,27 @@ func Stream(ctx context.Context, streamer Streamer) error {
 		}
 	}
 }
+
+// sleep returns a time to wait using random jitter and exponential backoff.
+func sleep(retries int) time.Time {
+	// waitMs := rand.Intn( 							// Get a random integer between 0 and ...
+	// 	min( 											// the minimum of ...
+	// 		streamerMaxFetchIntervalDuration,           // the max fetch interval and ...
+	// 		streamerFetchIntervalDuration*(1<<retries), // d*2^r, where r=retries and d= the normal
+	// 	),
+	// )
+	waitMs := rand.Intn(
```

Comment:
Do we just wait for 8s then 16s then 32s then 32s and so on?

Reply:
We wait for a random number of seconds between 0 and 4, then 0 and 8, 0 and 16, 0 and 32 for the maximum backoff. This smooths out the curve of retries from many simultaneous clients and means that the pattern of average backoffs is 2s, 4s, 8s, 16s.